src/code_190B20.c

```diff
@@ -103,22 +102,20 @@ s32 lookup_defense(DefenseTableEntry* defenseTable, Element elementKey) {
     DefenseTableEntry* row;
     s32 normalDefense = 0;
 
-    for (row = defenseTable; row->element != Element_END; row++) {
+    for (row = defenseTable; row->element != Element_END; row++, defenseTable++) {
         if (row->element == Element_NORMAL) {
-            normalDefense = row->defense;
+            normalDefense = defenseTable->defense;
```

Comment:
Is reusing defense table here necessary? Seems like the main change is the assignment and break below this. At the very least, can you try without the reuse?

Reply:
Yes it's required: https://i.imgur.com/MWxR6ql.png -- the problem with the function is that there are 2 iterators, both a0 (defenseTable) and v1 (row) and they increment differently and are used differently. a0 is used for the 2 loads inside the ifs, whereas v1 is used for the if conditions themselves.

Reply:
:+1:

Reply:
Very awkward that they did this; good job matching it!